pkg/utils/subst/subst.go

```diff
@@ -36,110 +42,156 @@ func ParseFile(file string, fss ...vfs.FileSystem) (SubstitutionTarget, error) {
 	return s, nil
 }
 
+var yqLibLogInit sync.Once
+
 func Parse(data []byte) (SubstitutionTarget, error) {
+	yqLibLogInit.Do(func() {
+		var lvl glog.Level
+		switch ocmlog.Context().GetDefaultLevel() {
```

Comment:
I'm not sure this is needed honestly. If you mock with the library you won't change the default logging level on the package side. You would change it for the context logger that your are using throughout your calls. For which this would have no effect at all.

I would just disable the logging completely for yq.

Alternatively, modify the parser to take an OCM context or a logger context that could be passed in and use that. Something like, WithLogger, or something. I doubt too many things use this as is, so it could be a fairly okay breaking change.

Reply:
I definitely don't want to disable the logging completely.   While it was chatty prior to this change it also turned out useful a couple of times.    Saved me from having to step through everything to understand what went wrong.

It does make sense to pass in the Context, however I really think that should be a separate PR.

Since the mapping would be needed once Context is passed in, I think my opinion is that 
- this mapping logic remain as is for now
- a ticket be created for changing the interface to take a Context 
- in the above ticket, mention that the mapping should be changed to map from the passed in Context when it is available

Reply:
👍